packages/mui-material/src/Autocomplete/Autocomplete.test.js

```diff
@@ -413,6 +413,46 @@ describe('<Autocomplete />', () => {
         expect(getAllByRole('button', { hidden: false })).to.have.lengthOf(5);
       }
     });
+
+    it('When the input box needs to expand downward, the listbox should remain open.', () => {
+      const options = [
+        'The Lord of the Rings: The Return of the King',
+        'The Good, the Bad and the Ugly',
+        'The Shawshank Redemption',
+        'Star Wars: Episode V - The Empire Strikes Back',
+      ];
+      const defaultValue = [
+        'The Lord of the Rings: The Return of the King',
+        'The Good, the Bad and the Ugly',
+        'The Shawshank Redemption',
+      ];
+      const handleClose = spy();
+      const handleOpen = spy();
+      const ref = React.createRef();
+      render(
+        <Autocomplete
+          multiple
+          onClose={handleClose}
+          onOpen={handleOpen}
+          limitTags={2}
+          options={options}
+          defaultValue={defaultValue}
+          renderInput={(params) => (
+            <TextField
+              {...params}
+              slotProps={{
+                input: { ...params.InputProps, ref },
+              }}
+              variant="standard"
+            />
+          )}
+        />,
+      );
+
+      fireEvent.mouseDown(ref.current);
```

Comment:
`mousedown` on `ref.current` means clicking the TextField's border, not the `input`. The `handleClose` call count should be once in the `next` branch, so this test isn't accurate. It's difficult to test this. I'll try other methods.

Reply:
The `input` was implemented with a `click event` in the `next branch`, but it has been changed to a `mouseDown` event in this branch. Will the function test code, rather than DOM tests, yield valid results in both branches? I tried various codes to simulate `mouseDown` the `TextField`, but both branches continued to show the same results, and I wasn't sure what to do next. I really want to solve this problem. I would appreciate an explanation after the modifications.

Reply:
I tried multiple ways. It is hard to simulate the behavior of the bug in the `next` branch in test to make the test fail on `next` and pass here.

Reply:
okay, I will closed this PR. Thank you!

Reply:
I thought it was happening in [PR](https://github.com/mui/material-ui/issues/42432), so I closed PR. sorry.

Can I change the way I fixed the bug and try writing the test code again? I'm not sure, but it might be a good idea to try another method.

Reply:
Feel free to.